recordval.go

```diff
@@ -74,98 +63,63 @@ func (r *Value) TypeOfColumn(col int) Type {
 	return TypeRecordOf(r.Type).Columns[col].Type
 }
 
-func (r *Value) Access(field string) (Value, error) {
-	col, ok := r.ColumnOfField(field)
+func (v *Value) Deref(field string) *Value {
+	if v == nil {
+		return nil
+	}
+	col, ok := v.ColumnOfField(field)
```

Comment:
This will panic if value is not a record (we want a nil instead right)?

Reply:
I pushed a fix for this based on your earlier comment.